chart/templates/hpa-external-metric.yaml

```
@@ -1,8 +1,12 @@
 {{- $root := . -}}
 {{- range $name, $service := $root.Values.web.webServices }}
 {{- if $service.autoscaling.enabled }}
-{{- $pool := "sum:puma.pool_capacity{kube_cluster_name:dsva-vagov-" }}
-{{- $threads := "sum:puma.max_threads{kube_cluster_name:dsva-vagov-" }}
+{{- $pool := "avg:puma.pool_capacity{kube_cluster_name:dsva-vagov-" }}
+{{- $pool_with_filters := "(" $pool $root.Values.target_env "-cluster,kube_deployment:" $name ")" }}
+{{- $threads := "avg:puma.max_threads{kube_cluster_name:dsva-vagov-" }}
+{{- $threads_with_filters := "(" $threads $root.Values.target_env "-cluster,kube_deployment:" $name ")" }}
```

What do you think of
```suggestion
{{- $max_threads := "avg:puma.max_threads{kube_cluster_name:dsva-vagov-" }}
{{- $max_available_threads := "(" $max_threads $root.Values.target_env "-cluster,kube_deployment:" $name ")" }}
```